setup.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# (C) British Crown Copyright 2013-2018 Met Office.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+
+import os
+from setuptools import setup
+# Monkey patching to disable version normalization, as we are using dates with leading zeroes
+# https://github.com/pypa/setuptools/issues/308
+from setuptools.extern.packaging import version
+
+version.Version = version.LegacyVersion
+from isodatetime import __version__
+
+
+# Utility function to read the README file.
+# Used for the long_description.  It's nice, because now 1) we have a top level
+# README file and 2) it's easier to type in the README file than to put a raw
+# string in below ...
+def read(fname):
+    return open(os.path.join(os.path.dirname(__file__), fname)).read()
+
+
+setup(
+    name="isodatetime",
+    version=__version__,
+    author="Met Office Modelling Infrastructure Support Systems",
+    author_email="metomi@metoffice.gov.uk",
+    description=("Python ISO 8601 date time parser and data model/manipulation utilities"),
+    license="GPL",
+    keywords="isodatetime datetime iso8601 date time parser",
+    url="https://github.com/metomi/isodatetime",
+    packages=['isodatetime'],
+    long_description=read('README.md'),
+    platforms='any',
+    install_requires=[],
+    python_requires='>=2.7',
+    classifiers=[
+        "Development Status :: 5 - Production/Stable",
+        "Environment :: Other Environment",
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+        "Operating System :: OS Independent",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3.6",
```

Comment:
I think the easiest is to have a clean break and go straight to Python 3 - given that this repository is unlikely to get a lot of development in the coming months. If necessary, we can create a release branch (from current release point) to back port essential fixes for Python 2.

Reply:
Making the move to Python 3 only would make it much easier :tada: 

Before trying to suggest any change in order to support Python 3, I will try to get the coverage of tests at least, so that it at least indicates where to look for possible issues when changing the code.

Reply:
Great!

setup_coverage => review_tests => python3

With Python3 integer division problems are particularly likely so lest make sure those are well covered.

Reply:
+1 ! :tada: will come back to this pull request after setup_coverage is done, and then start working on review_tests.